D/tests/syntax_test_old.d

```diff
@@ -58,30 +58,30 @@ template testTemplate(X)
 //                  ^ storage.attribute
 
 auto takeByRef(ref int h)
-// ^ storage.type
+// ^ storage.modifier
```

Comment:
Is is kind of an implicit type but behaves much more like an annotation:
You can use it with `:`
```
auto:
static foo() {}
```

It can be in any order:
```
auto @safe foo() {}
```

It isn't even required:
```
static foo() {}
```

Reply:
To add to some of the weirdness, auto is sometimes invalid as a type standin:
```
foreach (auto f; foo) {} // fails to compile
foreach (f; foo) {} // only this is valid
```